Fix GUI default onAppClose behavior on first run

JohnGahagan opened the pull request:
### **Description:**
This change ensures that the GUI provides non-null default values for (onAppClose) on first run.

Previously, the GuiSettingNotifier could return null for these settings on the first run, causing the GUI to close running instances automatically, even though the UI displayed "Ask" as the default.

With this fix:

- GuiSettingNotifier now returns explicit defaults for first-run scenarios.
- The runtime behavior now matches the defaults displayed in the GUI, improving consistency and preventing unexpected 
closures.

Impact:

- Resolves a bug where the GUI would close running instances immediately on first launch.
- Ensures first-run experience is aligned with UI expectations.

Resolves: #4506

tobe2098 commented:
@JohnGahagan , thank you. The CLA was a service failure. The message from the new commit should also follow the format, but the PR is otherwise complete.
Apparently the dart code also needs additional formatting as well.
Almost there!

JohnGahagan commented:
@tobe2098 Should be good to go now!

tobe2098 reviewed (changes requested):
Your solution is good and works well, but it repeats the return statement. Since we are introducing "default values', what do you think about this suggestion?

JohnGahagan commented:
Hi @tobe2098 done (I think I did it correctly).

I rebased and squashed the branch down to a single commit, updated the commit message to match the CONTRIBUTING format, and force-pushed the cleaned history to the PR branch. The previous “combination/merge” commits should be gone now.

If you’re able, could you re-run the checks / take another look? Thanks again!

tobe2098 commented:
Thank you @JohnGahagan , it is almost perfect now. The verify commits indicates you need to wrap the body at 72 chars. If you experience issues with commit messages I recommend you copy `git-hooks/commit-msg.py` to `.git/hooks/commit-msg` to have the checks run when you commit.
Apparently there is also some linting errors from dart. You can follow the indications from the workflow itself to resolve this.
